loopchain/blockchain/blockchain.py

```diff
@@ -591,8 +591,7 @@ def __add_block(self, block: Block, confirm_info, need_to_write_tx_info=True, ne
             if not need_to_write_tx_info:
                 receipts = None
 
-            if next_prep and self.find_preps_addresses_by_roothash(
-                    Hash32.fromhex(next_prep['rootHash'], ignore_prefix=True)):
+            if next_prep and self.find_preps_addresses_by_roothash(block.header.next_reps_hash):
```

Comment:
I think the way to compute the hash from next_prep itself is safer than using block's next_reps_hash.